include/circt/Dialect/HW/HWTypes.td

```diff
@@ -176,7 +176,8 @@ def TypeAliasType : HWType<"TypeAlias"> {
 
 /// An attribute to indicate the output file an operation should be emitted to.
 def OutputFileAttr : StructAttr<"OutputFileAttr", HWDialect, [
-  StructFieldAttr<"path", StrAttr>,
+  StructFieldAttr<"directory", OptionalAttr<StrAttr>>,
+  StructFieldAttr<"name", OptionalAttr<StrAttr>>,
```

Comment:
I'm fine with a split for directory and filename, another option would be to do just by string contents.  a name ending in a directory separator would be a unnamed file but named directory, and name without a directory separator would be a named file in the default directory, and something with both would have both.  This is fairly classic unix-y file v.s. directory v.s. file in a directory syntax.